### ISSUE TITLE
ElasticSearch cluster issues post Operator upgrade 

### ISSUE THREAD

**Issue by adasari** [elastic/cloud-on-k8s#1531]
## Bug Report
**What did you do?**
Performed following steps 
1. 
a. deployed operator 0.8.1
b. created ES (ElasticSearch) cluster (spec 7.2.0) with node 1
c. upgraded operator 0.9.0
d. upgraded ES cluster 7.3.0
and
2.
a. created operator 0.8.1
b. upgraded operator to 0.9.0
c. created ES cluster 

**What did you expect to see?**

Expected ES cluster upgraded or created without any issue.

**What did you see instead? Under which circumstances?**

ES cluster is not created(in Scenario 2)/upgraded (Scenario 1).

**Environment**

* Version information:

	version information is available in the steps performed.

* Kubernetes information:
	 On premise 
	
	
```
$ kubectl version

Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:44:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:32:14Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```

* Logs:
 Scenario 1:
```
{"level":"info","ts":1565264888.8940606,"logger":"elasticsearch-controller","msg":"Start reconcile iteration","iteration":1,"namespace":"default","es_name":"quickstart"}
{"level":"info","ts":1565264888.8943298,"logger":"annotation","msg":"Resource was previously reconciled by incompatible controller version and missing annotation, adding annotation","controller_version":"0.9.0","namespace":"default","name":"quickstart","kind":"Elasticsearch"}
{"level":"info","ts":1565264888.9046113,"logger":"elasticsearch-controller","msg":"End reconcile iteration","iteration":1,"took":0.01054987,"namespace":"default","es_ame":"quickstart"}
{"level":"error","ts":1565264888.9046648,"logger":"kubebuilder.controller","msg":"Reconciler error","controller":"elasticsearch-controller","request":"default/quickstart","error":"Internal error occurred: failed calling webhook \"validation.elasticsearch.elastic.co\": Post https://elastic-webhook-service.elastic-system.svc:443/validate-elasticsearches?timeout=30s: service \"elastic-webhook-service\" not found","stacktrace":"github.com/elastic/cloud-on-k8s/operators/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/elastic/cloud-on-k8s/operators/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217\ngithub.com/elastic/cloud-on-k8s/operators/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\ngithub.com/elastic/cloud-on-k8s/operators/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\ngithub.com/elastic/cloud-on-k8s/operators/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\ngithub.com/elastic/cloud-on-k8s/operators/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/elastic/cloud-on-k8s/operators/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
{"level":"info","ts":1565264888.9264312,"logger":"kubebuilder.webhook","msg":"starting the webhook server."}
{"level":"info","ts":1565264889.9053335,"logger":"elasticsearch-controller","msg":"Start reconcile iteration","iteration":2,"namespace":"default","es_name":"quickstart"}
{"level":"info","ts":1565264889.9056396,"logger":"annotation","msg":"Resource was previously reconciled by incompatible controller version and missing annotation, adding annotation","controller_version":"0.9.0","namespace":"default","name":"quickstart","kind":"Elasticsearch"}
{"level":"info","ts":1565264889.917695,"logger":"es-validation","msg":"ValidationHandler handler called","operation":"UPDATE","name":"quickstart","namespace":"default"}
{"level":"info","ts":1565264889.931914,"logger":"elasticsearch-controller","msg":"End reconcile iteration","iteration":2,"took":0.026581773,"namespace":"default","es_ame":"quickstart"}
{"level":"info","ts":1565264889.9320426,"logger":"elasticsearch-controller","msg":"Start reconcile iteration","iteration":3,"namespace":"default","es_name":"quickstart"}
{"level":"info","ts":1565264889.9321115,"logger":"license-controller","msg":"Start reconcile iteration","iteration":2,"namespace":"default","es_name":"quickstart"}
{"level":"info","ts":1565264889.9320993,"logger":"annotation","msg":"Resource was created with older version of operator, will not take action","controller_version":"0.9.0","resource_controller_version":"0.8.0-UNKNOWN","namespace":"default","name":"quickstart"}
{"level":"info","ts":1565264889.9323258,"logger":"elasticsearch-controller","msg":"End reconcile iteration","iteration":3,"took":0.000283968,"namespace":"default","es_ame":"quickstart"}
{"level":"info","ts":1565264889.9323125,"logger":"license-controller","msg":"Re-queuing new license check","namespace":"default","es_name":"quickstart","RequeueAfter":3600}
{"level":"info","ts":1565264889.9323423,"logger":"license-controller","msg":"End reconcile iteration","iteration":2,"took":0.000231445,"namespace":"default","es_name":"quickstart"}
```
  Scenario 2 
```
{"level":"info","ts":1565267890.0980039,"logger":"license-validation","msg":"ValidationHandler handler called","operation":"CREATE","name":"","namespace":"default"}
{"level":"info","ts":1565267890.1013277,"logger":"generic-reconciler","msg":"Creating resource","kind":"Secret","namespace":"default","name":"quickstart-es-elastic-user"}
{"level":"info","ts":1565267890.1029212,"logger":"license-validation","msg":"ValidationHandler handler called","operation":"CREATE","name":"","namespace":"default"}
{"level":"info","ts":1565267890.3627362,"logger":"generic-reconciler","msg":"Creating resource","kind":"Secret","namespace":"default","name":"quickstart-es-xpack-file-realm"}
{"level":"info","ts":1565267890.3665793,"logger":"license-validation","msg":"ValidationHandler handler called","operation":"CREATE","name":"","namespace":"default"}
{"level":"info","ts":1565267890.4727728,"logger":"observer","msg":"Creating observer for cluster","namespace":"default","es_name":"quickstart"}
E0808 12:38:10.477438       1 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:196: Failed to list *v1beta1.PodDisruptionBudget: poddisruptionbudgets.policy is forbidden: User "system:serviceaccount:elastic-system:elastic-operator" cannot list resource "poddisruptionbudgets" in API group "policy" at the cluster scope
E0808 12:38:11.480532       1 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:196: Failed to list *v1beta1.PodDisruptionBudget: poddisruptionbudgets.policy is forbidden: User "system:serviceaccount:elastic-system:elastic-operator" cannot list resource "poddisruptionbudgets" in API group "policy" at the cluster scope
E0808 12:38:12.483651       1 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:196: Failed to list *v1beta1.PodDisruptionBudget: poddisruptionbudgets.policy is forbidden: User "system:serviceaccount:elastic-system:elastic-operator" cannot list resource "poddisruptionbudgets" in API group "policy" at the cluster scope
E0808 12:38:13.495877       1 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:196: Failed to list *v1beta1.PodDisruptionBudget: poddisruptionbudgets.policy is forbidden: User "system:serviceaccount:elastic-system:elastic-operator" cannot list resource "poddisruptionbudgets" in API group "policy" at the cluster scope
E0808 12:38:14.497991       1 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:196: Failed to list *v1beta1.PodDisruptionBudget: poddisruptionbudgets.policy is forbidden: User "system:serviceaccount:elastic-system:elastic-operator" cannot list resource "poddisruptionbudgets" in API group "policy" at the cluster scope
E0808 12:38:15.502062       1 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:196: Failed to list *v1beta1.PodDisruptionBudget: poddisruptionbudgets.policy is forbidden: User "system:serviceaccount:elastic-system:elastic-operator" cannot list resource "poddisruptionbudgets" in API group "policy" at the cluster scope
```

**Comment by thbkrkr** [comment]
Hello,

As described in https://www.elastic.co/guide/en/cloud-on-k8s/0.9/release-highlights-0.9.0.html#k8s_breaking_changes, existing Elasticsearch and Kibana resources that were created with a 0.8.x version of the operator are not compatible with the 0.9.0 operator.

This explains why the Elasticsearch cluster is not upgraded in your scenario 1 and the presence of the log: `Resource was created with older version of operator, will not take action","controller_version":"0.9.0","resource_controller_version":"0.8.0-UNKNOWN"`.

The non creation of the Elasticsearch cluster in your scenario 2 is odd. The log `Failed to list *v1beta1.PodDisruptionBudget: poddisruptionbudgets.policy is forbidden: User "system:serviceaccount:elastic-system:elastic-operator" cannot list resource "poddisruptionbudgets" in API group "policy" at the cluster scope` indicates there is an issue with a missing RBAC rule.

What's the output of the followinf `kubectl` command for you?
```shell
> kubectl -n elastic-system get clusterrole elastic-operator -o json | \
    jq '.rules[] | select(.apiGroups[0]=="policy")' 
{
  "apiGroups": [
    "policy"
  ],
  "resources": [
    "poddisruptionbudgets"
  ],
  "verbs": [
    "get",
    "list",
    "watch",
    "create",
    "update",
    "patch",
    "delete"
  ]
}
```

**Comment by adasari** [comment]
i don't see any apiGroups with policy. looks my way of upgrading es-operator is incorrect. let me try again. thanks

**Comment by adasari** [comment]
i upgraded StatefulSet and ClusterRole now. Scenario 2 looks good. Thank you.
